_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+
+## Deploy 
+
+In the previous step we created an image which we can use now to deploy our application. We are going to create another build configuration: **Deploy**. This build will run a Docker container on the **TeamCity** agent based on the image from the **Release** phase.
+
+Our build configuration must contain three elements. The first one is a trigger. Again, we use the **Finish Build Trigger** with a dependency to the **Release** build configuration.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_trigger.png" alt="Deploy Trigger" title="Deploy Trigger" class="img img-responsive style-screengrab">
+
+The second element is a version of our image. We can obtain this information from the **artifact** created by the **Release** build. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_artifact.png" alt="Deploy Artifact" title="Deploy Artifact" class="img img-responsive style-screengrab">
+
+The last step is a little bit more complicated. Before we deploy the new version of the application the deployment script need to check if the container with the previous version is already deployed on the machine. If we have such a container we need to stop and remove it. 
+Next we read the current version from the **artifact** and create the new container based this version.  
+
+```
+if [ -n "$(docker ps --filter label=simple_application -q)" ]; then 
+   docker stop simple_application
+   docker rm simple_application
+fi;
+
+version_to_deploy=$(cat image.version)
+docker run -d -p 80:4567 \
+           --name simple_application \
+           --label simple_application \
+           registry.private/simple_application:${version_to_deploy}
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_step.png" alt="Deploy Step" title="Deploy Step" class="img img-responsive style-screengrab">
+
+If we now run a build of our application and we pass all tests the **Release** and **Deploy** builds, we will start an automatic deployment of our application.
+ 
+Continuous Deployment is ready.
+
+## It's done.
+
+The continuous deployment pipeline described in this post is of course simplified. Between our **Release** and **Deploy** steps we would like to do some 
+additional tests on a production-like environment or introduce [zero downtime deployments](http://codurance.com/services/training/devops-training/), but our approach to any deployment should remain unchanged. 
+
+When we go outside of our code we should use our application image. The image expresses a set of our application behaviours which are consistent for each version of the image. Now you also decide how your code is executed and you cannot blame admins anymore for installing a wrong version of Java or Ruby.
```

Comment:
- When we go outside of our code we should use our application image (I'm not sure I understand what you mean by "going outside of our code"). 
- The image expresses a set of our application behaviours which are consistent for each version of the image (This is also not so clear to me)

- Now you also decide how your code is executed and you cannot blame admins anymore for installing a wrong version of Java or Ruby. (I would replace this with: "As a developer, you are now not only in charge of how your application is executed but also your production environment and how your application is deployed. With great power comes great responsibility and you cannot put the blame on your system admins anymore. You are the one in control now. ")